File: src/main/java/aoc2021/Day1.kt
package aoc2021

fun main() = Day1().run()


class Day1 : Day(1) {
    override fun part1(input: List<String>) : Any{
        var count = 0
        for (i in 1 until input.size) {
            if (input[i].toInt() > input[i - 1].toInt())
                count++
        }
        return count
    }

    override fun part2(input: List<String>) : Any{
        var count = 0
        for (i in 0 until input.size - 3) {
            if (input[i].toInt() < input[i + 3].toInt())
                count++
        }
        return count
    }
}

File: src/main/java/aoc2022/Day1.scala
package aoc2022

import scala.collection.mutable.ListBuffer
object Day1 extends Day(1) {
  def main(args: Array[String]): Unit = Day1.run()

  val meals = new ListBuffer[Int]()

  val currentElf = new ListBuffer[Int]()
  input.foreach(line => if (line.isEmpty) {
    meals.addOne(currentElf.sum)
    currentElf.clear()
  } else currentElf.addOne(line.toInt))


  override def part1(): Any = meals.max

  override def part2(): Any = meals.sorted.takeRightInPlace(3).sum

}
